src/main/java/org/threadly/concurrent/future/SettableListenableFuture.java

```diff
@@ -131,7 +169,9 @@ public void setFailure(Throwable failure) {
     synchronized (resultLock) {
       this.failure = failure;
       
-      setDone();
+      if (! setDone()) {
```

Comment:
Ah yes, I will fix this.  Though it absolutely will make an atomic boolean impossible, since we don't want threads blocking on .get() to wake up before the result has been set (and notify happens in setDone() right now)